test/test_fsdp2/test_fsdp2_eager.py

```diff
@@ -74,8 +76,16 @@ def get_local_inp(self, dtype: torch.dtype = torch.float32):
         dist.broadcast(global_inp, src=0)
         return global_inp.view(self.world_size, -1)[self.rank].view(16, 16)
 
-    def swap_linear_with_dynamic(self, module: nn.Module, **kwargs: Any) -> nn.Module:
-        return swap_linear_with_float8_linear(module, Float8DynamicLinear, **kwargs)
+    def swap_linear_with_dynamic(
```

Comment:
Maybe losing some context but is there a reason why the existing swap function doesnt work?

Reply:
if the question is why do we need `swap_linear_with_dynamic`, we probably don't.  Removing that is not related to this PR though so I left it for a future person.